src/main/java/com/mraof/minestuck/client/gui/ColorSelectorScreen.java

```diff
@@ -2,130 +2,187 @@
+			redSlider.setValue((color >> 16) & 0xFF);
+			greenSlider.setValue((color >> 8) & 0xFF);
+			blueSlider.setValue(color & 0xFF);
+			hexBox.setValue(toFormattedHex(color));
 		}
-		//Alpha colors
-		for(int i = 0; i < 4; i++)
+		else if(tab==Tab.RGB)
 		{
-			int color = ColorHandler.getColor(i + 4) | 0xFF000000;
-			int x = 21 + 34*i;
-			fill(poseStack, xOffset + x, yOffset + 53, xOffset + x + 32, yOffset + 69, color);
+			hexBox.setValue(toFormattedHex(getSlidersValue()));
 		}
-		//Troll colors
-		for(int xIndex = 0; xIndex < 4; xIndex++)
-			for(int yIndex = 0; yIndex < 3; yIndex++)
-			{
-				int color = ColorHandler.getColor(yIndex*4 + xIndex + 8) | 0xFF000000;
-				int x = 21 + 34*xIndex;
-				int y = 74 + 18*yIndex;
-				fill(poseStack, xOffset + x, yOffset + y, xOffset + x + 32, yOffset + y + 16, color);
-			}
+		
+		tabButton.setMessage(tab==Tab.RGB?Component.translatable(BASIC_TAB):Component.translatable(ADVANCED_TAB));
 	}
 	
-	private void renderSelectionBox(PoseStack poseStack)
+	@Override
+	public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTicks)
 	{
-		int xOffset = (width - guiWidth)/2;
-		int yOffset = (height - guiHeight)/2;
+		renderBackground(poseStack);
+		
+		RenderSystem.setShader(GameRenderer::getPositionTexShader);
+		RenderSystem.setShaderColor(1, 1, 1, 1);
+		RenderSystem.setShaderTexture(0, guiBackground);
+		blit(poseStack, xOffset, yOffset, 0, 0, guiWidth, guiHeight);
 		
-		if(selectedIndex != -1)
+		String cacheMessage = I18n.get(SELECT_COLOR);
+		font.draw(poseStack, cacheMessage, (width / 2F) - font.width(cacheMessage) / 2F, yOffset + 12, 0x404040);
+		
+		if(tab==Tab.RGB)
 		{
-			int x = 19 + (selectedIndex % 4)*34;
-			int y = 30 + (selectedIndex /4)*18;
-			if(selectedIndex >= 4)
-				y += 3;
-			if(selectedIndex >= 8)
-				y += 3;
-			
+			// hide the color boxes by putting this other box on top of them
 			RenderSystem.setShader(GameRenderer::getPositionTexShader);
 			RenderSystem.setShaderColor(1, 1, 1, 1);
 			RenderSystem.setShaderTexture(0, guiBackground);
-			this.blit(poseStack, xOffset + x, yOffset + y, guiWidth, 0, 36, 20);
+			blit(poseStack, xOffset+20, yOffset+31, 0, guiHeight, 138, 98);
+			
+			// spirograph color preview
+			RenderSystem.setShaderColor((float)redSlider.getValue()/255F, (float)greenSlider.getValue()/255, (float)blueSlider.getValue()/255, 0.5F);
+			blit(poseStack, xOffset+106, yOffset+57, 47, 47, guiWidth, 20, 64, 64, 256, 256);
+		} else
+		{
+			for(ColorSelector canonColor : canonColors)
+			{
+				canonColor.draw(poseStack);
+				
+				// SELECTION BOX
+				if(selectedIndex == canonColor.id)
+				{
+					RenderSystem.setShader(GameRenderer::getPositionTexShader);
+					RenderSystem.setShaderColor(1, 1, 1, 1);
+					RenderSystem.setShaderTexture(0, guiBackground);
+					blit(poseStack, xOffset + canonColor.x - 2, yOffset + canonColor.y - 2, guiWidth, 0, ColorSelector.WIDTH + 4, ColorSelector.HEIGHT + 4);
+				}
```

Comment:
Instead of describing the code with a comment, you can describe it with a function name
```java
if(selectedIndex == canonColor.id)
	drawSelectionBox(canonColor);
```